platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -2,12 +2,40 @@ import merge from 'lodash.merge';
 import { CommandsManager } from '../../classes';
 import { ExtensionManager } from '../../extensions';
 import { PubSubService } from '../_shared/pubSubServiceInterface';
+import type { RunCommand, Commands } from '../../types/Command';
 
 const EVENTS = {
   TOOL_BAR_MODIFIED: 'event::toolBarService:toolBarModified',
   TOOL_BAR_STATE_MODIFIED: 'event::toolBarService:toolBarStateModified',
 };
 
+export type ButtonListeners = {
+  newStack?: RunCommand;
+  activeViewportIdChanged?: RunCommand;
+};
+
+export interface ButtonProps {
+  primary?: Button;
+  secondary?: Button;
+  items?: Button[];
+}
+
+export interface Button extends Commands {
+  id: string;
+  icon?: string;
+  label?: string;
+  type?: string;
+  tooltip?: string;
+  isActive?: boolean;
+  listeners?: ButtonListeners;
+  props?: ButtonProps;
+}
```

Comment:
ButtonListeners have cornerstone logic/event (newStack) which doesn't seem right. Not really sure what/where is the best way/place to add these Enums. any thoughts? 

PS: I think later on someone can add another listener for their custom tool doing some custom event, how are we handling that?